benchmarks/reframe_config.py

```diff
@@ -157,6 +157,39 @@ def spack_root_to_path():
                 },
             ]
         },  # end CSD3 CentOS 7
+        {
+            # https://www.rc.ucl.ac.uk/docs/Clusters/Kathleen/#node-types
+            'name': 'kathleen',
+            'descr': 'Kathleen',
+            'hostnames': ['login[0-9]+.kathleen.ucl.ac.uk'],
+            'max_local_jobs': 1,
+            'partitions': [
+                {
+                    'name': 'compute-node',
+                    'descr': 'Kathleen compute nodes',
+                    'scheduler': 'sge',
+                    'launcher': 'mpirun',
+                    'environs': ['default'],
+                    'max_jobs': 36,
+                    'processor': {
+                        'num_cpus': 80,
+                        'num_cpus_per_core': 2,
+                        'num_sockets': 2,
+                        'num_cpus_per_socket': 20,
+                    },
+                    'resources': [
+                        {
+                            'name': 'mpi',
+                            'options': ['-pe mpi {num_slots}'],
+                        },
+                        {
+                            'name': 'threads',
+                            'options': ['-l threads=2'], # disable hyperthreading (default). To enable, use threads=1
```

Comment:
Can you do this on the command line? Editing this file to toggle this is not ideal.

Reply:
ok, I misunderstood how the resource should be declared/used. How's this:

```suggestion
                            'options': ['-l threads={virtual_cpus_per_thread}'], # default is 2 to disable hyperthreading. To enable, use {'threads' : {'virtual_cpus_per_thread': 1}}
```

Reply:
The above is the value for the `self.extra_resources` inside a benchmark that would want to enable hyperthreading.

Reply:
Maybe this should be controlled via [`use_multithreading`](https://reframe-hpc.readthedocs.io/en/stable/regression_test_api.html#reframe.core.pipeline.RegressionTest.use_multithreading)?  At the moment the SGE scheduler doesn't do anything special in ReFrame about that, but if desired we could do something like https://github.com/reframe-hpc/reframe/blob/2b289c55e1c1015dd7f2dd04fdeac33f3a30305d/reframe/core/schedulers/slurm.py#L199-L202 for Slurm (as long as SGE has some generic enough configurations for this).

Reply:
I agree it would be great if this could be handled by ReFrame and controlled by `use_multithreading` in a benchmark.